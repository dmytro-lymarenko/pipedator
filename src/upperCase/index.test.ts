import { upperCase } from './index';

describe('upperCase()', () => {
	it('should validate a string in uppercase', () => {
		expect(upperCase().validate('')).toBeNull();
		expect(upperCase().validate('TEST THIS STRING')).toBeNull();
	});

	it('should fail', () => {
		expect(upperCase().validate('TEST This STRING')).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(upperCase('Custom message').validate('TEST This STRING')).toMatchSnapshot();
	});
});
