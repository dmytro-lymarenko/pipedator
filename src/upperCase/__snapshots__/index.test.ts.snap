// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upperCase() should fail 1`] = `
Object {
  "children": null,
  "message": "Value should be in uppercase",
  "path": "",
}
`;

exports[`upperCase() should support custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
